app/src/processing/app/Base.java

```diff
@@ -1313,6 +1350,63 @@ public void rebuildExamplesMenu(JMenu menu) {
   private static String priorPlatformFolder;
   private static boolean newLibraryImported;
 
+  public void selectTargetBoard(TargetBoard targetBoard) {
+    for (int i = 0; i < boardMenu.getItemCount(); i++) {
+      JMenuItem menuItem = boardMenu.getItem(i);
+      if (!(menuItem instanceof JRadioButtonMenuItem)) {
+        continue;
+      }
+
+      JRadioButtonMenuItem radioButtonMenuItem = ((JRadioButtonMenuItem) menuItem);
+      if (targetBoard.getName().equals(radioButtonMenuItem.getText())) {
+        radioButtonMenuItem.setSelected(true);
+        break;
+      }
+    }
+
+    BaseNoGui.selectBoard(targetBoard);
+    filterVisibilityOfSubsequentBoardMenus(boardsCustomMenus, targetBoard, 1);
+
+    onBoardOrPortChange();
+    rebuildImportMenu(Editor.importMenu);
+    rebuildExamplesMenu(Editor.examplesMenu);
+    try {
+      rebuildRecentBoardsMenu();
+    } catch (Exception e) {
+      // fail silently
+    }
+  }
+
+  public void rebuildRecentBoardsMenu() throws Exception {
+
+    Enumeration<AbstractButton> btns = recentBoardsButtonGroup.getElements();
+    while (btns.hasMoreElements()) {
+      AbstractButton x = btns.nextElement();
+      if (x.isSelected()) {
+        return;
+      }
+    }
+    btns = recentBoardsButtonGroup.getElements();
+    while (btns.hasMoreElements()) {
+      AbstractButton x = btns.nextElement();
+      boardMenu.remove(x);
+    }
```

Comment:
Why all this bookkeeping? It seems that `rebuildRecentBoardsMenu()` is only called from `rebuildBoardsMenu()`, which AFAICS starts with making `boardsMenu()` empty, so you can always just start from an empty slate here? Maybe that can be emphasized by renaming to `addRecentBoardsMenuItems()` or something like that, if you're worried about people calling this method from other contexts as well?